sim/examples/random-player-ai.ts

```diff
@@ -29,53 +24,15 @@ export class RandomPlayerAI extends BattlePlayer {
 		this.move = options.move || 1.0;
 		this.mega = options.mega || 0;
 		this.prng = options.seed && !Array.isArray(options.seed) ? options.seed : new PRNG(options.seed);
-
-		this.disabled = new Map();
-		this.trapped = new Set();
-		this.retry = false;
 	}
 
 	receiveError(error: Error) {
-		if (error.message.startsWith(`[Invalid choice]`) && this.retry) {
-			this.retry = false;
-			// If we get a choice error, we retry the choice using the last request provided
-			// we've got a '|callback' updating our state regarding what Pokemon are trapped
-			// or disabled.
-			this.makeChoice(this.lastRequest!);
-		} else {
-			throw error;
-		}
+		// If we made an unavailable choice we will receive a followup request to
+		// allow us the opportunity to correct our decision.
+		if (!error.message.startsWith('[Unavailable choice]')) throw error;
```

Comment:
```suggestion
		if (error.message.startsWith('[Unavailable choice]')) return;
		throw error;
```

Really minor nitpick, but I think this makes the code more readable.